src/api/user/user.controller.ts

```diff
@@ -25,4 +25,26 @@ export class UserController {
       result: userData,
     };
   }
+
+  @Get(':id')
+  async getExample(@Param('id') id: string): Promise<any> {
+    const result = await this.userService.findExampleById({ _id: id });
+
+    // Simulate non-optimized code
+    const modifiedResult = this.processResult(result);
+
+    return modifiedResult;
+  }
+
+  // Non-optimized code
+  private processResult(result: any): any {
+    let modifiedResult = result;
+
+    // Example non-optimized code
+    for (let i = 0; i < modifiedResult.length; i++) {
+      modifiedResult[i].name = modifiedResult[i].name.toUpperCase();
+    }
+
+    return modifiedResult;
+  }
```

Comment:
**Issue**: The newly added 'getExample' method and 'processResult' method contain non-optimized code that converts user names to uppercase one by one in a loop. This could be inefficient, especially with large datasets. <br> **Fix**: Consider using a more efficient approach such as map function or other batch processing to handle transformations, which can leverage modern JavaScript engine optimizations. <br> **Code Suggestion**: 
 ```
 - for (let i = 0; i < modifiedResult.length; i++) {
 -   modifiedResult[i].name = modifiedResult[i].name.toUpperCase();
 - }
 + modifiedResult = modifiedResult.map(user => ({
 +   ...user,
 +   name: user.name.toUpperCase()
 + }));
 ```